### PR TITLE
Add sphinx documentation

### PR DESCRIPTION
**Pull request by ribdp**
## Pull Request Description

Migrate documentation from markdown and mkdocs, to rst and Sphinx
Add script to be run for code documentation
Update the documentation workflow to build the sphinx docs

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [ ] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation by running CI/gen_doc/docs/gen_sysobj_doc.m script locally, if applies.

### CODE REVIEW

**Comment by ribdp** [comment]
Hi all, any comments here?

File: docs/source/reference_api/gen_sysobj_doc.m [hunk 1]
```diff
@@ -70,10 +70,16 @@
             all_props = [all_props,s];
         end
         top_doc = help(dotmfilename);
+        disp("Hello")
```
[the remark this answers is not in the document]
Comment (ribdp) [reply docs/source/reference_api/gen_sysobj_doc.m]: I think that was an older commit wherein I had added that line as a debug step. It's been removed in one of the subsequent commits. 

The commit history is honestly not the cleanest. I'll squash the commits in the end before merging.

**Comment by ribdp** [comment]
Hi @jansunil , @SaikiranGudla , @CMinajigi  - any further comments?

**Comment by ribdp** [comment]
Just fyi, you can check out the built documentation here - https://analogdevicesinc.github.io/PrecisionToolbox/

**Review by jansunil — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]

**Review by jansunil — APPROVED** [review]

**Pull request merged**